src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/ExtractOriginalAlignmentRecordsByNameSpark.java

```diff
@@ -0,0 +1,86 @@
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.StructuralVariationSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+
+/**
+ * Created by shuang on 3/3/17.
+ */
+@CommandLineProgramProperties(summary="Find original alignment records that have the requested read names and outputs them.",
+        oneLineSummary="Dump original alignment records that have the requested read names.",
+        usageExample = "ExtractOriginalAlignmentRecordsByNameSpark \\" +
+                "-I /path/to/my/dir/allAlignments.sam \\" +
+                "-O /path/to/my/dir/extractedOriginalAlignments.sam \\" +
+                "--readNameFile /path/to/my/dir/readNames.txt",
+        omitFromCommandLine = true,
+        programGroup = StructuralVariationSparkProgramGroup.class)
+@BetaFeature
+public final class ExtractOriginalAlignmentRecordsByNameSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+    private final Logger localLogger = LogManager.getLogger(ExtractOriginalAlignmentRecordsByNameSpark.class);
+
+    @Argument(doc = "file containing list of read names", fullName = "readNameFile")
+    private String readNameFile;
+
+    @Argument(doc = "file to write reads to", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    private String outputSAM;
+
+    @Argument(doc = "to require RG tag on reads or not [false]", shortName = "rg",
+            fullName = "requireRG", optional = true)
+    private boolean require = false;
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx ) {
+
+        final Broadcast<HashSet<String>> namesToLookForBroadcast = ctx.broadcast(parseReadNames());
+
+        final JavaRDD<GATKRead> reads =
+                getUnfilteredReads().repartition(80)
+                        .filter(read -> namesToLookForBroadcast.getValue().contains(read.getName())).cache();
+        localLogger.info("Found these many alignments: " + reads.count());
+
+        writeReads(ctx, outputSAM, reads);
+    }
+
+    private HashSet<String> parseReadNames() {
+
+        try ( final BufferedReader rdr =
+                      new BufferedReader(new InputStreamReader(BucketUtils.openFile(readNameFile))) ) {
+            final HashSet<String> namesToLookFor = new HashSet<>();
+            String line;
+            while ( (line = rdr.readLine()) != null ) {
+                namesToLookFor.add(line.replace("@", "")
```

Comment:
Is there a guaranteed that such sequences @ /1 /2 cannot be in the middle of a read name? Otherwise you could use a regex ("^@") and  ("/\d$").

Reply:
perhaps then you would call replaceAll or replaceFirst instead.

Reply:
Also you could use BufferedReader's ```Stream<String> lines()```:

rdr.lines().map(s -> s.replace(...).replace(...)).collect(Collectors.toSet()); 


But remember to capture UncheckedIOException.